### DISCUSSION TITLE
Why OpenAI API is very slow than ChatGPT

### DISCUSSION THREAD

**Discussion by kientv** [FlowiseAI/Flowise#1098] [General]
![image](https://github.com/FlowiseAI/Flowise/assets/3217438/87493177-5fc1-4bfa-a9fe-ecfd340dc30e)

![image](https://github.com/FlowiseAI/Flowise/assets/3217438/1557c5d8-4d4c-4f06-ae69-1bb9c16130c5)

**Comment by kientv** [comment]
Why we need this chain, because it's only re-define question or translate question.
How to turn-off it to faster?
![image](https://github.com/FlowiseAI/Flowise/assets/3217438/29137dee-3631-41c0-9025-0fa546109106)

**Reply by AnasAli986** [reply]
How are you able to look at this screen? Is it langsmith? How do I set it up? Thanks

**Reply by kientv** [reply]
Yes, it's langsmith. You need setup langsmith env only. Maybe you need referral code to regist an account.

**Comment by HenryHengZJ** [comment]
Conversational Retrieval QA Chain block is seperated into 2 chains:

1. Reformat question using conversation history
2. Use the retrieved context to answer the reformatted question

You can take a look at the template - Prompt Chaining with VectorStore, its what executed under the hood

**Reply by kientv** [reply]
@HenryHengZJ  Thank you.
I have just test "Prompt Chaining with VectorStore" without re-format question phrase.
It's much faster than. But LLMchain is not support Short Memory and "VectorStore to document" Search from vector store not working (every score).

![Screenshot_2023-10-21-13-09-33-17](https://github.com/FlowiseAI/Flowise/assets/3217438/5998b6ce-08c8-4274-8f8b-c7437a0b2ad4)
